web-console/knikubevirt/VMIs/VMIs.md

```diff
@@ -0,0 +1,60 @@
+# Surfacing VMIs in the UI
+
+VMI is the current running VM.
+The initial reason for surfacing VMIs in the UI was that there is no point in hiding the VMIs from the UI.
+
+## UCs
+
+1. Debugging - If a user thinks that something isn’t right with the VMs and he wants to check what’s going on. When a VMI isn’t managed by a VM and it’s invisible in the UI but that VMI is actually still there because it’s impossible to delete VMIs (that’s a basic Kubernetes mechanism). This can happen by an orphan delete, or that a replica set isn’t used.
+User can delete VMs and orphan the underlying VMIs so that the VM will get deleted but the VMI will still be kept.
+2. Users can write controllers for VMIs if they need specific UCs which aren’t covered by VMs. F.e. Users would never see VMIs because they are not shown.
+3. If in K8s we create Pods via Deployment, then why not create VMI via VMs so the user will learn K8s via VMs? Why create a gap between virtualization and K8s?
+4. Upstream Kubernetes is popular because in VMI Replica set and machine controller implementation that is used to “spork(?)” Kubernetes clusters on top of CNV
+Bare metal machines installed with Kubernetes
+Users install Kubevirt for CNV
+On these VMs there are clusters that can be scaled up and down and they create the VMIs (and not the VMs)
+
+## 2 Types of personas
+
+1. Traditional VM operation admin that is used to a certain view and behavior of the console, in a traditional virtualization way/ A user who has a virtualization background and has no need to understand the underlying k8s technology
+2. Users that use the console in a more ‘Kubernetes’ manner
+We need to try to educate users and narrow the gap for CNV users to better understand K8s/OpenShift.
+We want to allow this to be surfaced in the UI and live side by side in a way that will not confuse/ frighten/ intimidate the traditional user.
+
+## Mockups
+
+We had finalized into this option and will show 2 parallel columns in the VMs list page, one for each: VMs beside VMIs.
+
+![list of VMIs beside VMs](img/VMsListW_VMIsOp1.png)
+
+Trying to educate users about VMIs, we will provide hints, whenever it is appropriate. 
+In most usual cases, when no VM or VMI is missing, the instance name link will take the user directly to the instance page (VM/VMI).
+When a VMI is missing an owner VM, the user will be provided with an explanation in a popover for the reason it is missing. The name of the VM + badge will be shown in a disabled state for the sake of accessibility so a screen reader would be able to announce it.
+
+![VMI is in an off status](img/Op1_Hint1.png)
```

Comment:
> but as far as I know this would be the first time that clicking a resource's blue text would sometimes reveal a popover instead of immediately taking the user to that resource's Details page.

+1 , also on rows where we have VMs the blue text will immediately take the user to that resource's Details page, making two different behaviour patterns on the same list.

Maybe we can ?
1 - Do not link to VMIs if we have a VM, because if we have a VM user do need to interact with the VMI directly.
     Users can get to the VMI using the VM's detail page. 
2 - Show the VMIs that are owned by VMIRS ( e.g. "Orphan VMI" ) in a separate page.

Reply:
+1 
Updated the popover to connect with the greyed "No VM" and not to the blue resource.